### PR TITLE
Allow environment variables in the uri

### PR DESCRIPTION
**Pull request by zencodeuk**
Fix for issue https://github.com/tusharm/go-artifactory-plugin/issues/18 

Allows the uri to contain go env variables and be substituted at run time

### CODE REVIEW

File: src/main/java/com/tw/go/plugins/artifactory/model/GoArtifactFactory.java [hunk 1]
```diff
@@ -34,8 +36,11 @@ public GoArtifact apply(File file) {
             }
 
             private String artifactUri(String artifactName) {
-                return config.isFolder() ? config.uri() + "/" + artifactName : config.uri();
+                String uri = config.isFolder() ? config.uri() + "/" + artifactName : config.uri();
+                StrSubstitutor sub = new StrSubstitutor(context.environment().asMap(), "${", "}");
```
Comment (tusharm) [inline src/main/java/com/tw/go/plugins/artifactory/model/GoArtifactFactory.java]: We can skip the args "${" and "}"..they are the defaults.

Comment (tusharm) [inline src/main/java/com/tw/go/plugins/artifactory/model/GoArtifactFactory.java]: This is good stuff, thx!

**Comment by tusharm** [comment]
This is good stuff, thx!

**Comment by zencodeuk** [comment]
@tusharm updated to remove defaults :+1:

**Pull request merged**